wallet/src/de/schildbach/wallet/ui/send/SendCoinsOfflineTask.java

```diff
@@ -51,14 +51,23 @@ public SendCoinsOfflineTask(final Wallet wallet, final Handler backgroundHandler
     }
 
     public final void sendCoinsOffline(final SendRequest sendRequest) {
+        sendCoinsOffline(sendRequest, false);
+    }
+
+    public final void sendCoinsOffline(final SendRequest sendRequest, final boolean txAlreadyCompleted) {
         backgroundHandler.post(new Runnable() {
             @Override
             public void run() {
                 org.bitcoinj.core.Context.propagate(Constants.CONTEXT);
 
                 try {
                     log.info("sending: {}", sendRequest);
-                    final Transaction transaction = wallet.sendCoinsOffline(sendRequest); // can take long
+                    if (txAlreadyCompleted) {
+                        wallet.commitTx(sendRequest.tx);
+                    } else {
+                        wallet.sendCoinsOffline(sendRequest);
+                    }
```

Comment:
Where does the value of txAlreadyComplete come from? The logic here seems to be that if the tx is completed (`wallet.completeTx` was called) then it only needs to be committed, otherwise `wallet.sendCoinsOffline` will complete and commit the transaction.  I suppose this makes sense, so this relates to a comment above about how the value of txAlreadyComplete is set.